crates/topos-tce-storage/src/tests/mod.rs

```diff
@@ -368,13 +374,71 @@ async fn get_source_head_for_subnet(storage: RocksDBStorage) {
     .unwrap();
     storage.persist(&other_certificate_2, None).await.unwrap();
 
-    let last_certificate_subnet = storage
+    let last_certificate_subnet_2 = storage
         .get_source_heads(vec![SOURCE_SUBNET_ID_1])
         .await
         .unwrap()
         .last()
         .unwrap()
         .clone();
-    assert_eq!(other_certificate_2.id, last_certificate_subnet.cert_id);
-    assert_eq!(11, last_certificate_subnet.position.0); //check position
+    assert_eq!(other_certificate_2.id, last_certificate_subnet_2.cert_id);
+    assert_eq!(11, last_certificate_subnet_2.position.0); //check position
+}
+
+#[rstest]
+#[test(tokio::test)]
+async fn get_pending_certificates(storage: RocksDBStorage) {
+    let mut expected_pending_certificates: Vec<(u64, Certificate)> = Vec::new();
+    let mut expected_pending_certificates_count: u64 = 0;
+
+    let certificates_for_source_subnet_1 =
+        create_certificate_chain(SOURCE_SUBNET_ID_1, &[TARGET_SUBNET_ID_2], 15);
+    for cert in &certificates_for_source_subnet_1[0..10] {
+        storage.persist(cert, None).await.unwrap();
+    }
+    for cert in &certificates_for_source_subnet_1[10..] {
+        storage.add_pending_certificate(cert).await.unwrap();
+        expected_pending_certificates.push((expected_pending_certificates_count, cert.clone()));
+        expected_pending_certificates_count += 1;
+    }
+
+    let certificates_for_source_subnet_2 =
+        create_certificate_chain(SOURCE_SUBNET_ID_2, &[TARGET_SUBNET_ID_2], 15);
+    for cert in &certificates_for_source_subnet_2[0..10] {
+        storage.persist(cert, None).await.unwrap();
+    }
+    for cert in &certificates_for_source_subnet_2[10..] {
+        storage.add_pending_certificate(cert).await.unwrap();
+        expected_pending_certificates.push((expected_pending_certificates_count, cert.clone()));
+        expected_pending_certificates_count += 1;
+    }
```

Comment:
```suggestion
    let certificates_for_source_subnet_1 =
        create_certificate_chain(SOURCE_SUBNET_ID_1, &[TARGET_SUBNET_ID_2], 15);
    let certificates_for_source_subnet_2 =
        create_certificate_chain(SOURCE_SUBNET_ID_2, &[TARGET_SUBNET_ID_2], 15); 
  
    // Persist the first 10 Cert of each Subnets
    for cert in &certificates_for_source_subnet_1[0..10] {
        storage.persist(cert, None).await.unwrap();
    }
    for cert in &certificates_for_source_subnet_2[0..10] {
        storage.persist(cert, None).await.unwrap();
    }
    
    // Add the last 5 cert of each Subnet as pending certificate
    for cert in &certificates_for_source_subnet_1[10..] {
        storage.add_pending_certificate(cert).await.unwrap();
        expected_pending_certificates.push((expected_pending_certificates_count, cert.clone()));
        expected_pending_certificates_count += 1;
    }
    for cert in &certificates_for_source_subnet_2[10..] {
        storage.add_pending_certificate(cert).await.unwrap();
        expected_pending_certificates.push((expected_pending_certificates_count, cert.clone()));
        expected_pending_certificates_count += 1;
    }
```

suggestion to make the intention clearer, at least to add 1~2 comments to understand quicker what's the  setup